raiden/network/pathfinding.py

```diff
@@ -157,6 +158,42 @@ def get_pfs_info(url: str) -> Optional[PFSInfo]:
         return None
 
 
+def get_valid_pfs_url(
+    service_registry: ServiceRegistry,
+    index_in_service_registry: int,
+    block_identifier: BlockSpecification,
+) -> Optional[str]:
+    """Returns the URL for the PFS identified by the given index
+
+    Checks validity of registration in the ServiceRegistry contract and
+    checks the price from the PFS' info endpoint.
+
+    Returns PFS URL or None (if any check fails).
+    """
+    address = service_registry.ever_made_deposits(
+        block_identifier=block_identifier, index=index_in_service_registry
+    )
+    if not address:
+        return None
+
+    if not service_registry.has_valid_registration(
+        address=address, block_identifier=block_identifier
+    ):
+        return None
+
+    url = service_registry.get_service_url(
+        block_identifier=block_identifier, service_hex_address=to_canonical_address(address)
+    )
+    if not url:
+        return None
+
+    pfs_info = get_pfs_info(url)
+    if pfs_info is None or pfs_info.price > DEFAULT_PATHFINDING_MAX_FEE:
```

Comment:
The comparison should be done with the actual max fee, not the default value.

Reply:
Ah yes, I forgot that we already allow changing this.